### DISCUSSION TITLE
Shared Photo Library

### DISCUSSION THREAD

**Discussion by steilerDev** [steilerDev/icloud-photos-sync#277] [Q&A] [answered]
From @txwireless in #224 

> I've been searching for a method to sync the shared library since all of the other projects out there haven't got it figured it yet. This looks promising. One other thing to note is the shared library is on my wife's account and it is shared with me. If I sync from my account it doesn't access the shared library even though I believe the record data is presented when I went through the various calls. Instead it will only sync (well attempt to given the stack error) on the account (my wife's) who created it. Is this by design or a refinement? Wanted to let you know just in case you hadn't tested in that scenario.

**Comment by steilerDev** [comment]
The assets from the shared library should all present in the `_Shared-Photos` folder of your photos library.

Unfortunately those shared asset will not be present in any of your folders - this capability is currently not exposed by the iCloud API

I however have not tested the scenario, where the shared library is owned by someone else - maybe there is an edge case I could not replicate yet.

**Comment by txwireless** [comment]
I previously was poking around into this with icloudpd and posted some info [here](https://github.com/icloud-photos-downloader/icloud_photos_downloader/issues/455#issuecomment-1385992756).  Long and short is I believe you can retrieve the shared library name from the private endpoint but then you have to change the endpoint to the shared one to actually download the assets.  Again, swinging above my competence level here but it appears this is what is happening.

**Comment by steilerDev** [comment]
@txwireless have you ever been able to complete the sync? Was the `_Shared-Photos` album populated by any chance?

**Reply by txwireless** [reply, marked as answer]
Yes and yes.  It took multiple runs but I have 78K items in my shared folder taking up 782G.  If you need me to send you specific data via PM, happy to.

**Reply by steilerDev** [reply]
Yeah - the initial sync is always a bit painful because of unreliability on Apple's side leading to unexpected behavior (wheree I might be a little bit conservative at the moment on how I'm handling things - it's a learning process)

SoI'm guessing this can be closed, as the Shared Library is synced, independent of ownership?
